demo/src/main/java/com/eis/geoCalendar/demo/Behaviour/NetworkEventMapBehaviour.java

```diff
@@ -0,0 +1,263 @@
+        }
+        return usersColors;
+    }
+
+    /**
+     * Called by a fragment of the application that returns the parameters to Create a new event and update the map
+     *
+     * @param pos         The position of the event to create
+     * @param description The description of the event to create
+     */
+    @Override
+    public void onEventReturn(LatLng pos, String description) {
+        GPSPosition eventPos = new GPSPosition(pos.latitude, pos.longitude);
+        NetworkEvent event = new GenericNetworkEvent<>(eventPos, description, myself);
+        if (onEventCreatedListeners != null && !onEventCreatedListeners.isEmpty())
+            for (OnEventCreatedListener listener : onEventCreatedListeners) {
+                listener.onEventCreated(event);
+            }
+
+        Marker created = addMyEvent(event);
+        created.setTag(event);
+        moveMap(pos);
+        currentEvents.put(created, event);
+    }
+
+    /**
+     * Adds the given event to the map, with the custom icon if set, if the the default icon will be used
+     *
+     * @param myEvent An event created from the current user
+     * @return The marker that represents the event on the map
+     */
+    private Marker addMyEvent(NetworkEvent myEvent) {
+        LatLng pos = new LatLng(myEvent.getPosition().getLatitude(),
+                myEvent.getPosition().getLongitude());
+        if (userIcon != null)
+            return mMap.addMarker(new MarkerOptions().position(pos).icon
+                    (BitmapDescriptorFactory.fromBitmap(userIcon)).title(myEvent.getContent().toString()));
+        else
+            return mMap.addMarker(new MarkerOptions().position(pos).title(myEvent.getContent().toString()));
+        //automatically cuts title if too long
+    }
+
+    @Override
+    public void subscribeOnEventCreatedListener(@NonNull OnEventCreatedListener listener) {
+        if (onEventCreatedListeners == null)
+            onEventCreatedListeners = new ArrayList<>();
+        onEventCreatedListeners.add(listener);
+    }
+
+    @Override
+    public void unsubscribeOnEventCreatedListener(OnEventCreatedListener listener) {
+        if (onEventCreatedListeners != null)
+            onEventCreatedListeners.remove(listener);
+    }
+
+    /**
+     * If set {@code True} it will be allowed for the user to remove from the map events coming
+     * from the network, listeners will be notified.
+     * If set to {@code False} it won't be allowed and if the user tries to remove a
+     * network event from the UI using whatever view nothing will happen, and listeners won't be notified
+     * <p>
+     * Whatever value this flag will be set, the user will always be able to delete its own events
+     *
+     * @param allow Boolean value that indicates will of the user to be able to delete network events
+     *              from the map
+     */
+    public void allowMapRemovalNetworkEvents(boolean allow) {
+        this.allowMapRemovalNetworkEvents = allow;
```

Comment:
I thought of leaving the choice to who utilizes this object, so that a user can ignore events if he wants. Maybe in the future, we could add a filter to avoid tracking events from certain users.

Reply:
Oh it's clear now, nice choice.